lerobot/common/datasets/factory.py

```diff
@@ -55,7 +55,11 @@ def make_dataset(cfg, split: str = "train") -> LeRobotDataset | MultiLeRobotData
             "strings to load multiple datasets."
         )
 
-    if isinstance(cfg.dataset_repo_id, str) and cfg.env.name not in cfg.dataset_repo_id:
+    if (
+        cfg.env.name != "dora"
+        and isinstance(cfg.dataset_repo_id, str)
+        and cfg.env.name not in cfg.dataset_repo_id
+    ):
```

Comment:
nit: If the linter will allow you, please separate filters (or at least add just the comment)
```suggestion
    # A soft check to warn if the environment matches the dataset. Don't check if we are using a real world env (dora) of multiple datasets.
    if cfg.env.name != "dora" and isinstance(cfg.dataset_repo_id, str):
        if cfg.env.name not in cfg.dataset_repo_id:
    
```